src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -377,6 +387,117 @@ public void WriteValue(string value)
             this.FlushIfBufferThresholdReached();
         }
 
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer(); 
+            this.bufferWriter.Write(doubleQuote.Slice(0, 1).Span);
+            this.Flush();
+
+            int chunkSize = 2048;
+
+            for (int i = 0; i < value.Length; i += chunkSize)
+            {
+                int remainingChars = Math.Min(chunkSize, value.Length - i);
+                bool isFinalBlock = (i + remainingChars) == value.Length;
+
+                ReadOnlySpan<char> chunk = value.Slice(i, remainingChars);
+
+                int firstIndexToEscape = NeedsEscaping(chunk);
+
+                Debug.Assert(firstIndexToEscape >= -1 && firstIndexToEscape < value.Length);
+
+                if (firstIndexToEscape != -1)
+                {
+                    WriteEscapedStringChunk(chunk, firstIndexToEscape, isFinalBlock);
+                }
+                else
+                {
+                    WriteStringChunk(chunk, isFinalBlock);
+                }
+
+                // Flush the buffer if needed
+                this.FlushIfBufferThresholdReached();
+            }
+
+            this.bufferWriter.Write(doubleQuote.Slice(0, 1).Span);
+
+            // since we bypass the Utf8JsonWriter, we need to signal to other
+            // Write methods that a separator should be written first
+            if (this.isWritingAtStartOfArray || this.isWritingConsecutiveRawValuesAtStartOfArray || !this.IsInArray())
+            {
+                this.shouldWriteSeparator = true;
+            }

```

Comment:
I'd suggest putting this logic in a method and reusing it here and in `WriteRawValueCore` (we'll also need to use it when writing large byte arrays).